addon/mixins/route.js

```diff
@@ -193,6 +199,7 @@ const RouteMixin = Ember.Mixin.create({
       currentPage: startingPage,
       _firstPageLoaded: false,
       _perPage: perPage,
+      _store: store,
```

Comment:
Interesting. I have never used a store that wasn't just Ember DS extended. Indeed this functionality is something that someone might want. But I'd consider using non-Ember-DS datastore with nondefault query mechanisms a separate PR/feature (request) once this is in.